src/validation.js

```diff
@@ -59,18 +64,50 @@ export function validateField(fieldUri, options) {
   );
 }
 
+/**
+ * Returns a list of Statements for each validation in the form graph that is attached to the given field
+ *
+ * @param {NamedNode} fieldUri
+ * @param {{store: ForkingStore, formGraph: NamedNode, severity?: NamedNode}} options
+ * @returns {Statement[]}
+ */
 export function validationsForField(fieldUri, options) {
-  const { store, formGraph } = options;
-  const v2Result = store.match(
+  const { store, formGraph, severity } = options;
+
+  /** @type {Statement[]} */
+  let validations = store.match(
     fieldUri,
     FORM("validatedBy"),
     undefined,
     formGraph
   );
-  if (v2Result.length > 0) {
-    return v2Result;
+
+  if (validations.length === 0) {
+    // v1 model fallback, for backwards compatibility
+    validations = store.match(
+      fieldUri,
+      FORM("validations"),
+      undefined,
+      formGraph
+    );
   }
-  return store.match(fieldUri, FORM("validations"), undefined, formGraph);
+
+  validations = validations.filter((statement) => {
+    const constraint = statement.object;
+    const severityType = store
```

Comment:
suggestion: change naming for this one to constraintSeverity and for severity to targetSeverity (maybe just locally for this one)? It makes it easier for me to understand which is which